Tools/scripts/run_coverage.py

```diff
@@ -0,0 +1,275 @@
+            subprocess.run([waf_light, "configure", "--debug"], check=True)
+            subprocess.run([waf_light], check=True)
+        except subprocess.CalledProcessError as err:
+            print("ERROR :")
+            print(err.cmd)
+            print(err.output)
+            exit(1)
+        self.progress("Build examples and vehicle binaries done !")
+
+    def run_full(self):
+        """Run full coverage on maximum of ArduPilot binaries and test functions."""
+        self.progress("Running full test suite...")
+        self.run_build()
+        self.init_coverage()
+        self.progress("Running tests")
+        SPEEDUP = 5
+        TIMEOUT = 14400
+        autotest = os.path.join(root_dir, "Tools/autotest/autotest.py")
+
+        self.progress("Running run.examples")
+        subprocess.run([autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "--speedup=" + str(SPEEDUP),
+                        "run.examples"])
+        self.progress("Running run.unit_tests")
+        subprocess.run(
+            [autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.unit_tests", "run.unit_tests"])
+        self.progress("Running test.Plane and test.QuadPlane")
+        subprocess.run([autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.Plane", "test.Plane",
+                        "test.QuadPlane"])
+        self.progress("Running test.Sub")
+        subprocess.run([autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.Sub", "test.Sub"])
+        self.progress("Running test.Copter")
+        subprocess.run([autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.Copter", "test.Copter"])
+        self.progress("Running test.Helicopter")
+        subprocess.run(
+            [autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.Helicopter", "test.Helicopter"])
+        self.progress("Running test.Tracker")
+        subprocess.run(
+            [autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.Tracker", "test.Tracker"])
+        self.progress("Running test.Rover")
+        subprocess.run([autotest, "--timeout=" + str(TIMEOUT), "--debug", "--no-clean", "build.Rover", "test.Rover"])
+
+        # TODO add any other execution path/s we can to maximise the actually
+        # used code, can we run other tests or things?  Replay, perhaps?
+        self.update_stats()
+
+    def update_stats(self):
+        """Update Coverage statistics only.
+
+        Assume that coverage tests have been run.
+        """
+        self.progress("Generating Coverage statistics")
+        with open(self.LCOV_LOG, 'a') as log_file:
+            # we cannot use subprocess.PIPE and result.stdout to get the output as it will be too long and trigger
+            # BlockingIOError: [Errno 11] write could not complete without blocking
+            # thus we ouput to temp file, and print the file line by line...
```

Comment:
Well, we can try it.  But my guess is that we'll hit it just on normal script output at some stage, so you'll need a plan B if this doesn't work out.